src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py

```diff
@@ -0,0 +1,400 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import datetime
+import dateutil.parser
+from azure.cli.core._util import CLIError
+from azure.cli.core.commands.arm import resource_id, is_valid_resource_id
+from ._client_factory import (get_devtestlabs_management_client)
+from .sdk.devtestlabs.models.gallery_image_reference import GalleryImageReference
+from azure.graphrbac import GraphRbacManagementClient
+import azure.cli.core.azlogging as azlogging
+
+logger = azlogging.get_az_logger(__name__)
+
+# pylint: disable=line-too-long
+
+# Odata filter for startswith
+ODATA_STARTS_WITH_FILTER = "startswith(name, '{}')"
+
+
+def validate_lab_vm_create(namespace):
+    """ Validates parameters for lab vm create and updates namespace. """
+    formula = None
+
+    collection = [namespace.image, namespace.formula]
+    if not _single(collection):
```

Comment:
there's not really any reason you couldn't chose a formula and override its image